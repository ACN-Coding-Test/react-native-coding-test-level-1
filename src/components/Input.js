import React from "react";
import PropTypes from "prop-types";
import { StyleSheet, TextInput } from "react-native";

import theme from "../theme";

function Input(props) {
  const { value, inputStyles, ...otherProps } = props;
  return (
    <TextInput
      value={value}
      {...otherProps}
      style={[styles.inputStyle, inputStyles]}
      textStyle={styles.textStyle}
      underlineColorAndroid={theme.colors.transparent}
    />
  );
}

Input.propTypes = {
  value: PropTypes.string,
};

const styles = StyleSheet.create({
  inputStyle: {
    width: "100%",
    borderWidth: 1,
    paddingHorizontal: theme.padding.default,
    paddingVertical: theme.padding.default,
    fontSize: theme.fontSizes.body,
    borderRadius: theme.borderRadius.default,
  },
});

export default Input;
