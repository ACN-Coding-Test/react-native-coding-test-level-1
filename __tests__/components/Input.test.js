import React from 'react'
import renderer from 'react-test-renderer'
import Input from '../../src/components/Input'

describe('Input:', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Input />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

