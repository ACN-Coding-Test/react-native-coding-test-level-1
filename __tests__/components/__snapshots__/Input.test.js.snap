// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input: should render correctly 1`] = `
<TextInput
  style={
    Array [
      Object {
        "borderRadius": 8,
        "borderWidth": 1,
        "fontSize": 20,
        "paddingHorizontal": 15,
        "paddingVertical": 15,
        "width": "100%",
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
